Userland/Libraries/LibCore/System.h

```diff
@@ -206,6 +210,57 @@ ErrorOr<void> unlockpt(int fildes);
 ErrorOr<void> access(StringView pathname, int mode);
 ErrorOr<DeprecatedString> readlink(StringView pathname);
 
+template<Pointer T>
+struct OwnPtrWithCustomDeleter {
```

Comment:
This seems like it could be generally useful? Why is it in Stream.h rather than AK?

Reply:
- I don't know if this is "good enough" to fit in AK. It should probably be a part of OwnPtr, and I'm not sure that maintainers will let *me* rewrite it to do so.
- AFAICT, the principal use case will be to interface with C types, and the best file to write this kind of interface is LibCore/System (not stream)

I will ask on Discord for a broader view

Reply:
Imo, I don't think it needs to be part of `OwnPtr` (I don't think you'd want to make all OwnPtr's pay the cost of a having a deleter member)

Reply:
There is probably a way to avoid it by with a `[[no_unique_address]]` and some template trickery, but it's not something I want to do for this PR.
Anyway, people on discord agrees with you, let me push that!